cogs/committee_actions_tracking.py

```diff
@@ -105,8 +205,161 @@ async def _create_action(
             raise InvalidActionDescriptionError(
                 message=DUPLICATE_ACTION_MESSAGE
             ) from create_action_error
+        await self._update_action_board()
         return action
 
+    @overload
+    @staticmethod
+    async def get_user_actions(
+        action_user: discord.Member | discord.User, status: list[str]
+    ) -> list[AssignedCommitteeAction]: ...
+
+    @overload
+    @staticmethod
+    async def get_user_actions(
+        action_user: "Iterable[discord.Member]",
+        status: list[str],
+    ) -> dict[discord.Member, list[AssignedCommitteeAction]]: ...
+
+    @staticmethod
+    async def get_user_actions(
+        action_user: "discord.Member | discord.User | Iterable[discord.Member]",
+        status: str | list[str],
+    ) -> list[AssignedCommitteeAction] | dict[discord.Member, list[AssignedCommitteeAction]]:
```

Comment:
Function signature can be simplified to always return a mapping with the selected user, even if `action_user` has type `discord.Member | discord.User`.

And use classmethods, rather than static methods (unless absolutely necessary), to allow calling other classmethods.

```suggestion
    @classmethod
    async def get_user_actions(
        cls,
        action_user: "discord.Member | discord.User | Iterable[discord.Member | discord.User]",
        status: "Collection[str]",
    ) -> "Mapping[discord.Member, Collection[AssignedCommitteeAction]]":
```